### PR TITLE
UINV-191 Implement Lock total invoice form logic

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UINV-191

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 1]
```diff
@@ -309,19 +324,48 @@ const InvoiceForm = ({
...
-                          type="checkbox"
+                          onChange={enableManualAmount}
                           vertical
                         />
                       </Col>
-                    </Row>
-
-                    <Row>
+                      <Col data-test-col-manual-amount xs={3}>
+                        {isEditPostApproval
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/InvoiceForm.js:342]: Isn't it handled inside of `TextField` with `isNonInteractive` prop?

File: src/invoices/InvoiceForm/InvoiceForm.test.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    expect(getByText('stripes-acq-components.FormFooter.cancel')).toBeDefined();
+    expect(getByText('ui-invoice.saveAndClose')).toBeDefined();
+  });
+
+  it('Than lock total should be unchecked and manual amount is read only', () => {
+    const { getByTestId } = renderInvoiceForm();
+
+    expect(getByTestId('lock-total').checked).toEqual(false);
+    expect(getByTestId('manual-amount')).toHaveAttribute('readonly');
+  });
```
Comment (axelhunn) [inline src/invoices/InvoiceForm/InvoiceForm.test.js:67]: Maybe split this into 2 tests?

File: src/invoices/InvoiceForm/InvoiceForm.test.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      });
+
+      const lockTotal = getByTestId('lock-total');
+
+      fireEvent(lockTotal, new MouseEvent('click', {
+        bubbles: true,
+        cancelable: true,
+      }));
+
+      expect(getByTestId('lock-total').checked).toEqual(true);
```
Comment (axelhunn) [inline src/invoices/InvoiceForm/InvoiceForm.test.js:86]: Should we test this?

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js:342]: Removed `isNonInteractive ` prop since we need to display amounts with currency sign

**Review by aliaksei-chumakou — APPROVED** [review]

**Review by axelhunn — APPROVED** [review]

File: src/invoices/InvoiceForm/InvoiceForm.test.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  });
+
+  it('should uncheck lock-total initially', () => {
+    const { getByTestId } = renderInvoiceForm();
+
+    expect(getByTestId('lock-total').checked).toEqual(false);
+  });
+
+  describe('When lock total is unchecked', () => {
+    it('then manual amout is readonly', async () => {
```
Comment (axelhunn) [inline src/invoices/InvoiceForm/InvoiceForm.test.js:69]:
```suggestion
    it('then manual amout is readonly', () => {
```

File: src/invoices/InvoiceForm/InvoiceForm.test.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    expect(getByText('ui-invoice.saveAndClose')).toBeDefined();
+  });
+
+  it('should uncheck lock-total initially', () => {
+    const { getByTestId } = renderInvoiceForm();
+
+    expect(getByTestId('lock-total').checked).toEqual(false);
+  });
+
+  describe('When lock total is unchecked', () => {
```
Comment (axelhunn) [inline src/invoices/InvoiceForm/InvoiceForm.test.js:68]: Maybe add explicit unchecking of 'lock-total', 'cause that's the case here basically.

**Pull request merged**